lib/cli/src/wasmer_env.rs

```diff
@@ -0,0 +1,127 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Error};
+use once_cell::sync::Lazy;
+use url::Url;
+use wasmer_registry::WasmerConfig;
+
+/// Command-line flags for determining the local "Wasmer Environment".
+///
+/// This is where you access `$WASMER_DIR`, the `$WASMER_DIR/wasmer.toml` config
+/// file, and specify the current registry.
+#[derive(Debug, Clone, PartialEq, clap::Parser)]
+pub struct WasmerEnv {
```

Comment:
I'd love to unify the Edge CLI ( which currently uses a very similar approach, but with different code).

Can we put this into the `registry` crate directly, maybe under a feature flag to hard required  require `clap` for compiling `registry`?

I realize that's a bit awkward, but putting it into `registry`, or another helper crate is the only way to allow the Edge CLI to use it, since the main CLI depends on the Edge CLI.

Reply:
The `wasmer-registry` crate is already a bit of a dumping ground. I'm sure nobody will notice if we add yet another chunky dependency to it 😅